### ISSUE TITLE
Evaluate synergies with Backstage service catalog

### ISSUE THREAD

**Issue by jmgrimes** [OpenSLO/OpenSLO#13]
Just had this pop onto my radar thanks to a colleague, and didn't know quite how to get this into discussion.  There's a current CNCF SIG for service catalogs called Backstage that seems to have a lot of synergy with this project.  In particular, the "service" construct here seems to have some strong alignment with the Component or Resource constructs in Backstage's model.  I wondered if it might be worth a look at supporting attachment of SLOs defined here to backstage components as well as the OpenSLO native service types.

**Comment by ian-bartholomew** [comment]
This is a great idea!

**Comment by weyert** [comment]
Yeah, I think it would be great to have a plugin for that! I have been talking with Backstage about it in the past; originally I had the idea to make it a new type of Service Catalog. But now I think it might be better to link to a project catalog entry.

**Comment by CRThaze** [comment]
One somewhat confusing thing is that Backstage calls a "service" and entity representing software you develop and deploy. While calling the grouping that provides APIs and is comprised of code you develop and the resources that support it, a "System".  To me, it feels right that SLOs be defined at the "System" level. But obviously, in the OpenSLO spec we group SLOs under an entity called a "service"...

**Comment by weyert** [comment]
Yeah, that's true. But I think `service` would make sense for SLOs though? So there is a bit of terminology mismatch between OpenSLO and Backstage but I don't think that would be a major problem. In theory you could introduce a `slo` entity type but if I understand correctly that would end in a database so you might have two copies of the SLO definition. Not sure if that's wanted

**Comment by proffalken** [comment]
Heh, this is really timely, I've just started exploring Backstage having used it at a client in the past.

At the moment, my plan for SLI/SLO integration is to use Sloth or similar to post the various alert rules into Prom/Cortex and then use the [Prometheus](https://roadie.io/backstage/plugins/prometheus/) and [Grafana](https://github.com/K-Phoen/backstage-plugin-grafana) plugins for Backstage to render them in the UI.

To have a custom component that pulled in the actual OpenSLO spec and displayed it alongside thresholds that are breached would be even better!

**Comment by CRThaze** [comment]
@proffalken Do we have an existing YAML Schema definition somewhere? I'm pretty new, but the only thing I'd seen for a formal (computer-readable) definition of the spec was in the Oslo Go structs. Is there somewhere else I should be looking?

**Comment by programmer04** [comment]
[OpenSLO/openslo-backstage-plugins](https://github.com/OpenSLO/openslo-backstage-plugins) is the home of the [Backstage](https://backstage.io) plugin, please discuss related issues, ideas, etc. in that repo
